File: examples/evaluation_order_demo/evaluation_order.py
obj = {}


def expr1():
    print("expr1")
    return obj


def expr2():
    print("expr2")
    return "answer"


def expr3():
    print("expr3")
    return 42


def expr4():
    print("expr4")
    return obj


def expr5():
    print("expr5")
    return "answer"


def expr6():
    print("expr6")
    return 1


def expr7(value1, value2, value3):
    print("expr7")


def expr8():
    print("expr8")
    return 8


def expr9():
    print("expr9")
    return 9


def expr10():
    print("expr10")
    return 10


print("Python: 'expr1()[expr2()] = expr3()'")
expr1()[expr2()] = expr3()
print(obj["answer"])

print("Python: 'expr4()[expr5()] += expr6()'")
expr4()[expr5()] += expr6()
print(obj["answer"])

print("Python: 'expr7(expr8(), expr9(), expr10())'")
expr7(expr8(), expr9(), expr10())

File: examples/evaluation_order_demo/evaluation_order.js
obj = {}

function expr1() {
  console.log("expr1");
  return obj;
}

function expr2() {
  console.log("expr2");
  return "answer";
}

function expr3() {
  console.log("expr3");
  return 42;
}

function expr4() {
  console.log("expr4");
  return obj;
}

function expr5() {
  console.log("expr5");
  return "answer";
}

function expr6() {
  console.log("expr6");
  return 1;
}

function expr7(value1, value2, value3) {
  console.log("expr7");
}

function expr8() {
  console.log("expr8");
  return 8;
}

function expr9() {
  console.log("expr9");
  return 9;
}

function expr10() {
  console.log("expr10");
  return 10;
}

console.log("JavaScript: 'expr1()[expr2()] = expr3()'");
expr1()[expr2()] = expr3();
console.log(obj.answer);

console.log("JavaScript: 'expr4()[expr5()] += expr6()'");
expr4()[expr5()] += expr6();
console.log(obj.answer);

console.log("JavaScript: 'expr7(expr8(), expr9(), expr10())'");
expr7(expr8(), expr9(), expr10());
